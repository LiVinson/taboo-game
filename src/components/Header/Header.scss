.header {
    background: none;
    font-family: 'Oswald', sans-serif;
    font-size: 3rem;
    color: var(--color-primary);
    text-shadow: 0px 1rem 1rem rgba(0,0,0,.6);
    text-transform:uppercase;
}

//For main homescreen only, display larger text.
.header--large {
    font-size: 6rem;
    text-align: center;
    margin: 1rem 0;
}



.subheading {
    font-size: 2.2rem;
    text-align: center;
    width: 30rem;
    margin: 0 auto;
    color: var( --color-gray-dark-1);

    font-weight: 400;
    font-family: 'Oswald', sans-serif;

}

.subheading__focus-text {
    color: var(--color-primary-light);
}
