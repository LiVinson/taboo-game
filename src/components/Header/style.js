import styled, { css } from 'styled-components'

export const StyledHeader = styled.header`
	margin-top: ${(props) => (props.large ? '3rem' : '2rem')};
	margin-bottom: ${(props) => (props.subheading ? '5rem' : '3rem')};
`

export const Title = styled.h1`
	background: none;
	font-family: ${(props) => props.theme.text.display};
	font-size: ${(props) => props.theme.fontSize.medium};
	color: ${(props) => props.theme.color.primary};
	text-transform: uppercase;
  font-weight: 300;

	${(props) =>
		props.large &&
		css`
			font-size: ${(props) => props.theme.fontSize.xlarge};
			text-align: center;
			margin: 1rem 0;
		`}
`
Title.displayName = 'Title'

export const Subheading = styled.h3`
  font-size: 2.4rem;
  text-align: center;
  width: 30rem;
  margin: 0 auto;
  font-weight: 300;
  color: ${(props) => props.theme.color.grayDark1};
  background-color: rgba(247, 249, 249, .5);
  line-height:1.3;
  
`
Subheading.displayName = 'Subbeading'

export const FocusSpan = styled.span`
	color: ${(props) => props.theme.color.primary};

`
FocusSpan.displayName = 'FocusSpan'
