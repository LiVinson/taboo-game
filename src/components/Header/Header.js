import React from "react"
import PropTypes from "prop-types"
import {  StyledHeader, Title, Subheading, FocusSpan} from "./style.js"


export default function Header({ location }) {
  //size of header and whether subheading included varies based on route location
  const homeOrEndRoute = location.pathname.includes("home") || location.pathname.includes("end")
  const homeRouteExact = location.pathname === ("/home")

  return (
    <StyledHeader large={homeOrEndRoute} subheading={homeRouteExact}>
      <Title large={homeOrEndRoute}>Taboo!</Title>
      {homeRouteExact && (
        <Subheading>
          The team game that’s all about what you{" "}
          <FocusSpan>say,</FocusSpan> and what you{" "}
          <FocusSpan>don’t!</FocusSpan>
        </Subheading>
      )}
    </StyledHeader>
  )
}

Header.propTypes = {
  location: PropTypes.object.isRequired,
}
