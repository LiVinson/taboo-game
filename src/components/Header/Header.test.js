import React from "react"
import { shallow, mount } from "enzyme"
import renderer from "react-test-renderer"
import Header from "./Header"

it("Renders large header correctly", () => {
  const props = {
    location: {
      pathname: "/home",
    },
  }
  const HeaderComponent = renderer.create(<Header {...props} />).toJSON()
  expect(HeaderComponent).toMatchSnapshot()
})

it("Renders regular header correctly", () => {
  const props = {
    location: {
      pathname: "/play",
    },
  }
  const HeaderComponent = renderer.create(<Header {...props} />).toJSON()
  expect(HeaderComponent).toMatchSnapshot()
})

it("Receives a location prop with pathname property", () => {
  const props = {
    location: {
      pathname: "/home",
    },
  }

  const HeaderComponent = mount(<Header {...props} />)

  expect(HeaderComponent.prop("location")).toMatchObject({
    pathname: "/home",
  })
})

it("renders large header and subheading on home route", () => {
  const props = {
    location: {
      pathname: "/home",
    },
  }

  const HeaderComponent = shallow(<Header {...props} />)
  expect(HeaderComponent.find("h1.header--large")).toHaveLength(1)
  expect(HeaderComponent.find("h3.subheading")).toHaveLength(1)
})

it("renders reular header and no subheader on none-home route", () => {
  const props = {
    location: {
      pathname: "/play",
    },
  }

  const HeaderComponent = shallow(<Header {...props} />)
  expect(HeaderComponent.find("h1.header--large")).toHaveLength(0)
  expect(HeaderComponent.find("h3.subheading")).toHaveLength(0)
})
