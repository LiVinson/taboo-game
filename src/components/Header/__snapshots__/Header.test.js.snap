// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders large header correctly 1`] = `
<header>
  <h1
    className="header header--large"
  >
    Taboo!
  </h1>
  <h3
    className="subheading"
  >
    The team game that’s all about what you
     
    <span
      className="subheading__focus-text"
    >
      say,
    </span>
     and what you
     
    <span
      className="subheading__focus-text"
    >
      don’t!
    </span>
  </h3>
</header>
`;

exports[`Renders regular header correctly 1`] = `
<header>
  <h1
    className="header null"
  >
    Taboo!
  </h1>
</header>
`;
